src/adaptors/adaptor_common.c

```diff
@@ -23,150 +23,68 @@
-                          pn_tls_t            **tls_session,
-                          qd_log_source_t     *log_source,
-                          uint64_t             conn_id,
-                          bool                 is_listener,
-                          bool                *tls_has_output,
-                          const char          *protocols[])
-{
-    const char *role = is_listener ? "listener" : "connector";
-
-    qd_log(log_source, QD_LOG_INFO, "[C%"PRIu64"] %s %s configuring ssl profile %s", conn_id, role, config->name, config->ssl_profile_name);
-
-    do {
-        // find the ssl profile
-        assert(qd);
-        qd_connection_manager_t *cm = qd_dispatch_connection_manager(qd);
-        assert(cm);
-        qd_config_ssl_profile_t *config_ssl_profile = qd_find_ssl_profile(cm, config->ssl_profile_name);
-        if (!config_ssl_profile) {
-            qd_log(log_source, QD_LOG_ERROR, "[C%"PRIu64"] %s %s unable to find ssl profile %s", conn_id, config->name, config->ssl_profile_name);
-            break;
-        }
-
-        int res;
-        // First free, then create pn domain
-        if (*tls_config)
-            pn_tls_config_free(*tls_config);
-        *tls_config = pn_tls_config(is_listener ? PN_TLS_MODE_SERVER : PN_TLS_MODE_CLIENT);
-
-        if (! *tls_config) {
-            qd_log(log_source, QD_LOG_ERROR, "[C%"PRIu64"] %s %s unable to create tls domain for ssl profile %s", conn_id, role, config->name, config->ssl_profile_name);
-            break;
-        }
-
-        if (config_ssl_profile->ssl_trusted_certificate_db) {
-            res = pn_tls_config_set_trusted_certs(*tls_config, config_ssl_profile->ssl_trusted_certificate_db);
-            if (res != 0) {
-                qd_log(log_source, QD_LOG_ERROR, "[C%"PRIu64"] %s %s unable to set tls trusted certificates (%d)", conn_id, role, config->name, res);
-                break;
-            }
-        }
-
-        // Call pn_tls_config_set_credentials only if "certFile" is provided.
-        if (config_ssl_profile->ssl_certificate_file) {
-            res = pn_tls_config_set_credentials(*tls_config,
-                                                config_ssl_profile->ssl_certificate_file,
-                                                config_ssl_profile->ssl_private_key_file,
-                                                config_ssl_profile->ssl_password);
-            if (res != 0) {
-                qd_log(log_source, QD_LOG_ERROR, "[C%"PRIu64"] HTTP2 %s %s unable to set tls credentials (%d)", conn_id, role, config->name, res);
-                break;
-            }
-        }
-        else {
-            qd_log(log_source, QD_LOG_INFO, "[C%"PRIu64"] sslProfile %s did not provide certFile", conn_id, config->ssl_profile_name);
-        }
+//size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 4096;
+//size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 8192;
+//size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 16384;
+size_t QD_ADAPTOR_MAX_BUFFER_SIZE = 32768;
```

Comment:
The best way to do this is to use ALLOC_DEFINE_CONFIG to provide the size of the buffer memory - see buffer.c for an example